### DISCUSSION TITLE
[Bug] 当上下文中的字数较多时，上游服务器的响应速度较慢，此时会报告 504 错误

### DISCUSSION THREAD

**Discussion by ITOTI-Y** [lobehub/lobehub#1730] [General Question] [answered]
### 💻 系统环境

Windows

### 📦 部署环境

Vercel / Zeabur / Sealos

### 🌐 浏览器

Edge

### 🐛 问题描述

当上下文中的字数较多时，有些服务器会在处理结束后才返回响应，此时会报504错误

### 🚦 期望结果

希望能在web端提供设置超时时间的选项

[a comment by an automated account is not shown]

**Comment by fevrax** [comment]
当上游响应慢时，Lobe-chat持续等待。即使上游响应完成，Lobe-chat仍然会处于等待状态。

When the upstream response is slow, Lobe-chat continues to wait. Even when the upstream response is complete, Lobe-chat remains in a waiting state.



![image](https://github.com/lobehub/lobe-chat/assets/58434904/200ca079-3c5b-4903-9374-f74d423a47fe)
![image](https://github.com/lobehub/lobe-chat/assets/58434904/c76f0e63-8c13-4d9d-b3f1-f0020329173a)

**Comment by ITOTI-Y** [comment, marked as answer]
问题已经解决，当使用vercel部署时，vercel hobby计划的最长响应时间为10s，当超过10s时，会返回504错误
![image](https://github.com/lobehub/lobe-chat/assets/35416777/73217066-cb3a-44ca-81a9-c629b7ad9320)

[a comment by an automated account is not shown]

**Comment by bruceye777** [comment]
我改成60秒还是不行
![QQ_1727169708554](https://github.com/user-attachments/assets/58a19863-ca1f-4941-aaeb-e50cf2fdb89b)

**Reply by arvinxx** [reply]
o1 响应会超过60s
